src/main/java/org/ibp/api/rest/labelprinting/GermplasmLabelPrinting.java

```diff
@@ -0,0 +1,546 @@
+	}
+
+	@Override
+	List<LabelType> getAvailableLabelTypes(final LabelsInfoInput labelsInfoInput) {
+		final List<LabelType> labelTypes = new LinkedList<>();
+
+		final GermplasmSearchRequest germplasmSearchRequest = (GermplasmSearchRequest) this.searchRequestService
+			.getSearchRequest(labelsInfoInput.getSearchRequestId(), GermplasmSearchRequest.class);
+
+		final List<UserDefinedField> attributeTypes = this.germplasmSearchService.getGermplasmAttributeTypes(germplasmSearchRequest);
+		final List<UserDefinedField> nameTypes = this.germplasmSearchService.getGermplasmNameTypes(germplasmSearchRequest);
+
+		// Germplasm Details labels
+		labelTypes.add(GERMPLASM_DETAILS_FIXED_LABEL_TYPES);
+
+		// Pedigree labels
+		labelTypes.add(PEDIGREE_FIXED_LABEL_TYPES);
+
+		// Names labels
+		final LabelType namesType = new LabelType("Names", "Names");
+		namesType.setFields(new ArrayList<>());
+		namesType.getFields().addAll(nameTypes.stream()
+			.map(attr -> new Field(toKey(attr.getFldno()), attr.getFcode()))
+			.collect(Collectors.toList()));
+		labelTypes.add(namesType);
+
+		// Attribiutes labels
+		final LabelType attirbutesType = new LabelType("Attributes", "Attributes");
+		attirbutesType.setFields(new ArrayList<>());
+		attirbutesType.getFields().addAll(attributeTypes.stream()
+			.map(attr -> new Field(toKey(attr.getFldno()), attr.getFcode()))
+			.collect(Collectors.toList()));
+		labelTypes.add(attirbutesType);
+
+		return labelTypes;
+	}
+
+	@Override
+	LabelsData getLabelsData(
+		final LabelsGeneratorInput labelsGeneratorInput) {
+		// Get raw data
+		final Integer searchRequestId = labelsGeneratorInput.getSearchRequestId();
+		final GermplasmSearchRequest germplasmSearchRequest = (GermplasmSearchRequest) this.searchRequestService
+			.getSearchRequest(searchRequestId, GermplasmSearchRequest.class);
+
+		final Set<String> addedColumnsPropertyIds = new HashSet<>();
+
+		labelsGeneratorInput.getFields().forEach((listOfSelectedFields) ->
+			this.addingColumnToGermplasmSearchRequest(listOfSelectedFields, addedColumnsPropertyIds)
+		);
+
+		if (!StringUtils.isBlank(labelsGeneratorInput.getSortBy()) && !addedColumnsPropertyIds.contains(labelsGeneratorInput.getSortBy())) {
+			addedColumnsPropertyIds.add(labelsGeneratorInput.getSortBy());
+		}
+
+		germplasmSearchRequest.setAddedColumnsPropertyIds(addedColumnsPropertyIds.stream().collect(Collectors.toList()));
+
+		final List<Integer> listOfGermplasmDetailsAndPedrigreeIds = Stream.of(PEDIGREE_FIELD.values())
+			.map(PEDIGREE_FIELD::getId)
+			.collect(Collectors.toList());
+
+		listOfGermplasmDetailsAndPedrigreeIds.addAll(
+			Stream.of(GERMPLASM_DETAILS_FIELD.values()).map(GERMPLASM_DETAILS_FIELD::getId).collect(Collectors.toList()));
+
+		final List<Integer> selectedFieldIds =
+			labelsGeneratorInput.getFields().stream().flatMap(Collection::stream).collect(Collectors.toList());
+
+		final boolean haveNamesOrAttributes =
```

Comment:
rename by fieldsContainsNamesOrAttributes

Reply:
The one that was required to be renamed is boolean haveNamesOrAttributed. You renamed the list @cuenyad